backend/danswer/background/celery/celery.py

```diff
@@ -173,42 +248,50 @@ def _sync_document_batch(document_ids: list[str], db_session: Session) -> None:
 # Periodic Tasks
 #####
 @celery_app.task(
-    name="check_for_document_sets_sync_task",
+    name="schedule_tasks",
     soft_time_limit=JOB_TIMEOUT,
 )
-def check_for_document_sets_sync_task() -> None:
-    """Runs periodically to check if any document sets are out of sync
-    Creates a task to sync the set if needed"""
+def schedule_tasks() -> None:
+    """Runs periodically to check if any tasks should be run and runs them"""
+
     with Session(get_sqlalchemy_engine()) as db_session:
+        # manage document syncing task
         # check if any document sets are not synced
         document_set_info = fetch_document_sets(
             user_id=None, db_session=db_session, include_outdated=True
         )
         for document_set, _ in document_set_info:
-            if not document_set.is_up_to_date:
-                task_name = name_document_set_sync_task(document_set.id)
-                latest_sync = get_latest_task(task_name, db_session)
-
-                if latest_sync and check_live_task_not_timed_out(
-                    latest_sync, db_session
-                ):
-                    logger.info(
-                        f"Document set '{document_set.id}' is already syncing. Skipping."
-                    )
-                    continue
-
-                logger.info(f"Document set {document_set.id} syncing now!")
+            if should_sync_doc_set(document_set, db_session):
+                logger.info(f"Syncing the {document_set.name} document set")
                 sync_document_set_task.apply_async(
                     kwargs=dict(document_set_id=document_set.id),
                 )
 
+        # manage document pruning
+        all_cc_pairs = get_connector_credential_pairs(db_session)
```

Comment:
prefer this to be a separate task whcih only checks for document pruning. The reason here is to try and have as much isolation as possible between these jobs—a failure in the document set sync check should not cause pruning to not run